# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build gstreamer-build libvorbis-build orc-build pango-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

if pkg-config --exists alsa-lib-build; then
  ALSA=1
  REQUIRES="$REQUIRES alsa-lib-build"
fi

if pkg-config --exists tremor-build; then
  TREMOR=1
  REQUIRES="$REQUIRES tremor-build"
fi

# configure (ac-2.69; am-1.15; lt-2.4.2)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/fb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/fb/lib ./configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install tools/.libs/gst-device-monitor-1.0 $DESTDIR/fb/bin
install tools/.libs/gst-discoverer-1.0 $DESTDIR/fb/bin
install tools/.libs/gst-play-1.0 $DESTDIR/fb/bin
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/allocators
install -m 644 gst-libs/gst/allocators/allocators.h $DESTDIR/fb/include/gstreamer-1.0/gst/allocators
install -m 644 gst-libs/gst/allocators/gstdmabuf.h $DESTDIR/fb/include/gstreamer-1.0/gst/allocators
install -m 644 gst-libs/gst/allocators/gstfdmemory.h $DESTDIR/fb/include/gstreamer-1.0/gst/allocators
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/app
install -m 644 gst-libs/gst/app/app.h $DESTDIR/fb/include/gstreamer-1.0/gst/app
install -m 644 gst-libs/gst/app/gstappsink.h $DESTDIR/fb/include/gstreamer-1.0/gst/app
install -m 644 gst-libs/gst/app/gstappsrc.h $DESTDIR/fb/include/gstreamer-1.0/gst/app
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/audio.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/audio-channels.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/audio-enumtypes.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/audio-format.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/audio-info.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiobasesink.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiobasesrc.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiocdsrc.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudioclock.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiodecoder.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudioencoder.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiofilter.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudioiec61937.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiometa.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudioringbuffer.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiosink.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/gstaudiosrc.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -m 644 gst-libs/gst/audio/streamvolume.h $DESTDIR/fb/include/gstreamer-1.0/gst/audio
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/fft.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/gstfft.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/gstfftf32.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/gstfftf64.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/gstffts16.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -m 644 gst-libs/gst/fft/gstffts32.h $DESTDIR/fb/include/gstreamer-1.0/gst/fft
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/codec-utils.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/descriptions.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/encoding-profile.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/encoding-target.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/gstdiscoverer.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/gstpluginsbaseversion.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/install-plugins.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/missing-plugins.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/pbutils.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -m 644 gst-libs/gst/pbutils/pbutils-enumtypes.h $DESTDIR/fb/include/gstreamer-1.0/gst/pbutils
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/riff
install -m 644 gst-libs/gst/riff/riff.h $DESTDIR/fb/include/gstreamer-1.0/gst/riff
install -m 644 gst-libs/gst/riff/riff-ids.h $DESTDIR/fb/include/gstreamer-1.0/gst/riff
install -m 644 gst-libs/gst/riff/riff-media.h $DESTDIR/fb/include/gstreamer-1.0/gst/riff
install -m 644 gst-libs/gst/riff/riff-read.h $DESTDIR/fb/include/gstreamer-1.0/gst/riff
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtcpbuffer.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtp-enumtypes.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtpbaseaudiopayload.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtpbasedepayload.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtpbasepayload.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtpbuffer.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtpdefs.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtphdrext.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/gstrtppayloads.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -m 644 gst-libs/gst/rtp/rtp.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtp
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtsp-enumtypes.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtspconnection.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtspdefs.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtspextension.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtspmessage.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtsprange.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtsptransport.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/gstrtspurl.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -m 644 gst-libs/gst/rtsp/rtsp.h $DESTDIR/fb/include/gstreamer-1.0/gst/rtsp
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/sdp
install -m 644 gst-libs/gst/sdp/gstmikey.h $DESTDIR/fb/include/gstreamer-1.0/gst/sdp
install -m 644 gst-libs/gst/sdp/gstsdp.h $DESTDIR/fb/include/gstreamer-1.0/gst/sdp
install -m 644 gst-libs/gst/sdp/gstsdpmessage.h $DESTDIR/fb/include/gstreamer-1.0/gst/sdp
install -m 644 gst-libs/gst/sdp/sdp.h $DESTDIR/fb/include/gstreamer-1.0/gst/sdp
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/tag
install -m 644 gst-libs/gst/tag/gsttagdemux.h $DESTDIR/fb/include/gstreamer-1.0/gst/tag
install -m 644 gst-libs/gst/tag/gsttagmux.h $DESTDIR/fb/include/gstreamer-1.0/gst/tag
install -m 644 gst-libs/gst/tag/tag.h $DESTDIR/fb/include/gstreamer-1.0/gst/tag
install -m 644 gst-libs/gst/tag/xmpwriter.h $DESTDIR/fb/include/gstreamer-1.0/gst/tag
install -d $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/colorbalance.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/colorbalancechannel.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideodecoder.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideoencoder.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideofilter.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideometa.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideopool.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideosink.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/gstvideoutils.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/navigation.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-blend.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-chroma.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-color.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-converter.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-dither.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-enumtypes.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-event.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-format.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-frame.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-info.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-multiview.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-overlay-composition.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-resampler.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-scaler.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/video-tile.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/videoorientation.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -m 644 gst-libs/gst/video/videooverlay.h $DESTDIR/fb/include/gstreamer-1.0/gst/video
install -d $DESTDIR/fb/lib
install gst-libs/gst/allocators/.libs/libgstallocators-1.0.so.0.* $DESTDIR/fb/lib/libgstallocators-1.0.so.0
ln -sf libgstallocators-1.0.so.0 $DESTDIR/fb/lib/libgstallocators-1.0.so
install gst-libs/gst/app/.libs/libgstapp-1.0.so.0.* $DESTDIR/fb/lib/libgstapp-1.0.so.0
ln -sf libgstapp-1.0.so.0 $DESTDIR/fb/lib/libgstapp-1.0.so
install gst-libs/gst/audio/.libs/libgstaudio-1.0.so.0.* $DESTDIR/fb/lib/libgstaudio-1.0.so.0
ln -sf libgstaudio-1.0.so.0 $DESTDIR/fb/lib/libgstaudio-1.0.so
install gst-libs/gst/fft/.libs/libgstfft-1.0.so.0.* $DESTDIR/fb/lib/libgstfft-1.0.so.0
ln -sf libgstfft-1.0.so.0 $DESTDIR/fb/lib/libgstfft-1.0.so
install gst-libs/gst/pbutils/.libs/libgstpbutils-1.0.so.0.* $DESTDIR/fb/lib/libgstpbutils-1.0.so.0
ln -sf libgstpbutils-1.0.so.0 $DESTDIR/fb/lib/libgstpbutils-1.0.so
install gst-libs/gst/riff/.libs/libgstriff-1.0.so.0.* $DESTDIR/fb/lib/libgstriff-1.0.so.0
ln -sf libgstriff-1.0.so.0 $DESTDIR/fb/lib/libgstriff-1.0.so
install gst-libs/gst/rtp/.libs/libgstrtp-1.0.so.0.* $DESTDIR/fb/lib/libgstrtp-1.0.so.0
ln -sf libgstrtp-1.0.so.0 $DESTDIR/fb/lib/libgstrtp-1.0.so
install gst-libs/gst/rtsp/.libs/libgstrtsp-1.0.so.0.* $DESTDIR/fb/lib/libgstrtsp-1.0.so.0
ln -sf libgstrtsp-1.0.so.0 $DESTDIR/fb/lib/libgstrtsp-1.0.so
install gst-libs/gst/sdp/.libs/libgstsdp-1.0.so.0.* $DESTDIR/fb/lib/libgstsdp-1.0.so.0
ln -sf libgstsdp-1.0.so.0 $DESTDIR/fb/lib/libgstsdp-1.0.so
install gst-libs/gst/tag/.libs/libgsttag-1.0.so.0.* $DESTDIR/fb/lib/libgsttag-1.0.so.0
ln -sf libgsttag-1.0.so.0 $DESTDIR/fb/lib/libgsttag-1.0.so
install gst-libs/gst/video/.libs/libgstvideo-1.0.so.0.* $DESTDIR/fb/lib/libgstvideo-1.0.so.0
ln -sf libgstvideo-1.0.so.0 $DESTDIR/fb/lib/libgstvideo-1.0.so
install -d $DESTDIR/fb/lib/gstreamer-1.0
test $ALSA && install ext/alsa/.libs/libgstalsa.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/ogg/.libs/libgstogg.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/pango/.libs/libgstpango.so $DESTDIR/fb/lib/gstreamer-1.0
test $TREMOR && install ext/vorbis/.libs/libgstivorbisdec.so $DESTDIR/fb/lib/gstreamer-1.0
install ext/vorbis/.libs/libgstvorbis.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/adder/.libs/libgstadder.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/app/.libs/libgstapp.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audioconvert/.libs/libgstaudioconvert.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audiorate/.libs/libgstaudiorate.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audioresample/.libs/libgstaudioresample.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/audiotestsrc/.libs/libgstaudiotestsrc.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/encoding/.libs/libgstencodebin.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/gio/.libs/libgstgio.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/playback/.libs/libgstplayback.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/subparse/.libs/libgstsubparse.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/tcp/.libs/libgsttcp.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/typefind/.libs/libgsttypefindfunctions.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videoconvert/.libs/libgstvideoconvert.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videorate/.libs/libgstvideorate.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videoscale/.libs/libgstvideoscale.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/videotestsrc/.libs/libgstvideotestsrc.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/volume/.libs/libgstvolume.so $DESTDIR/fb/lib/gstreamer-1.0
install -d $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-allocators-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-app-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-audio-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-fft-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-pbutils-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-plugins-base-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-riff-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-rtp-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-rtsp-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-sdp-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-tag-1.0.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pkgconfig/gstreamer-video-1.0.pc $DESTDIR/fb/lib/pkgconfig

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/gst-plugins-base-build.pc << EOF
Name: gst-plugins-base
Version: 1.6.4
Description: Base GStreamer plugins and libraries
Requires: $REQUIRES

devel=\\
/fb/include/gstreamer-1.0/gst/allocators/allocators.h \\
/fb/include/gstreamer-1.0/gst/allocators/gstdmabuf.h \\
/fb/include/gstreamer-1.0/gst/allocators/gstfdmemory.h \\
/fb/include/gstreamer-1.0/gst/app/app.h \\
/fb/include/gstreamer-1.0/gst/app/gstappsink.h \\
/fb/include/gstreamer-1.0/gst/app/gstappsrc.h \\
/fb/include/gstreamer-1.0/gst/audio/audio.h \\
/fb/include/gstreamer-1.0/gst/audio/audio-channels.h \\
/fb/include/gstreamer-1.0/gst/audio/audio-enumtypes.h \\
/fb/include/gstreamer-1.0/gst/audio/audio-format.h \\
/fb/include/gstreamer-1.0/gst/audio/audio-info.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiobasesink.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiobasesrc.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiocdsrc.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudioclock.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiodecoder.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudioencoder.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiofilter.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudioiec61937.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiometa.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudioringbuffer.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiosink.h \\
/fb/include/gstreamer-1.0/gst/audio/gstaudiosrc.h \\
/fb/include/gstreamer-1.0/gst/audio/streamvolume.h \\
/fb/include/gstreamer-1.0/gst/fft/fft.h \\
/fb/include/gstreamer-1.0/gst/fft/gstfft.h \\
/fb/include/gstreamer-1.0/gst/fft/gstfftf32.h \\
/fb/include/gstreamer-1.0/gst/fft/gstfftf64.h \\
/fb/include/gstreamer-1.0/gst/fft/gstffts16.h \\
/fb/include/gstreamer-1.0/gst/fft/gstffts32.h \\
/fb/include/gstreamer-1.0/gst/pbutils/codec-utils.h \\
/fb/include/gstreamer-1.0/gst/pbutils/descriptions.h \\
/fb/include/gstreamer-1.0/gst/pbutils/encoding-profile.h \\
/fb/include/gstreamer-1.0/gst/pbutils/encoding-target.h \\
/fb/include/gstreamer-1.0/gst/pbutils/gstdiscoverer.h \\
/fb/include/gstreamer-1.0/gst/pbutils/gstpluginsbaseversion.h \\
/fb/include/gstreamer-1.0/gst/pbutils/install-plugins.h \\
/fb/include/gstreamer-1.0/gst/pbutils/missing-plugins.h \\
/fb/include/gstreamer-1.0/gst/pbutils/pbutils.h \\
/fb/include/gstreamer-1.0/gst/pbutils/pbutils-enumtypes.h \\
/fb/include/gstreamer-1.0/gst/riff/riff.h \\
/fb/include/gstreamer-1.0/gst/riff/riff-ids.h \\
/fb/include/gstreamer-1.0/gst/riff/riff-media.h \\
/fb/include/gstreamer-1.0/gst/riff/riff-read.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtcpbuffer.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtp-enumtypes.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtpbaseaudiopayload.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtpbasedepayload.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtpbasepayload.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtpbuffer.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtpdefs.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtphdrext.h \\
/fb/include/gstreamer-1.0/gst/rtp/gstrtppayloads.h \\
/fb/include/gstreamer-1.0/gst/rtp/rtp.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtsp-enumtypes.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtspconnection.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtspdefs.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtspextension.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtspmessage.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtsprange.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtsptransport.h \\
/fb/include/gstreamer-1.0/gst/rtsp/gstrtspurl.h \\
/fb/include/gstreamer-1.0/gst/rtsp/rtsp.h \\
/fb/include/gstreamer-1.0/gst/sdp/gstmikey.h \\
/fb/include/gstreamer-1.0/gst/sdp/gstsdp.h \\
/fb/include/gstreamer-1.0/gst/sdp/gstsdpmessage.h \\
/fb/include/gstreamer-1.0/gst/sdp/sdp.h \\
/fb/include/gstreamer-1.0/gst/tag/gsttagdemux.h \\
/fb/include/gstreamer-1.0/gst/tag/gsttagmux.h \\
/fb/include/gstreamer-1.0/gst/tag/tag.h \\
/fb/include/gstreamer-1.0/gst/tag/xmpwriter.h \\
/fb/include/gstreamer-1.0/gst/video/colorbalance.h \\
/fb/include/gstreamer-1.0/gst/video/colorbalancechannel.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideodecoder.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideoencoder.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideofilter.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideometa.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideopool.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideosink.h \\
/fb/include/gstreamer-1.0/gst/video/gstvideoutils.h \\
/fb/include/gstreamer-1.0/gst/video/navigation.h \\
/fb/include/gstreamer-1.0/gst/video/video.h \\
/fb/include/gstreamer-1.0/gst/video/video-blend.h \\
/fb/include/gstreamer-1.0/gst/video/video-chroma.h \\
/fb/include/gstreamer-1.0/gst/video/video-color.h \\
/fb/include/gstreamer-1.0/gst/video/video-converter.h \\
/fb/include/gstreamer-1.0/gst/video/video-dither.h \\
/fb/include/gstreamer-1.0/gst/video/video-enumtypes.h \\
/fb/include/gstreamer-1.0/gst/video/video-event.h \\
/fb/include/gstreamer-1.0/gst/video/video-format.h \\
/fb/include/gstreamer-1.0/gst/video/video-frame.h \\
/fb/include/gstreamer-1.0/gst/video/video-info.h \\
/fb/include/gstreamer-1.0/gst/video/video-multiview.h \\
/fb/include/gstreamer-1.0/gst/video/video-overlay-composition.h \\
/fb/include/gstreamer-1.0/gst/video/video-resampler.h \\
/fb/include/gstreamer-1.0/gst/video/video-scaler.h \\
/fb/include/gstreamer-1.0/gst/video/video-tile.h \\
/fb/include/gstreamer-1.0/gst/video/videoorientation.h \\
/fb/include/gstreamer-1.0/gst/video/videooverlay.h \\
/fb/lib/libgstallocators-1.0.so \\
/fb/lib/libgstapp-1.0.so \\
/fb/lib/libgstaudio-1.0.so \\
/fb/lib/libgstfft-1.0.so \\
/fb/lib/libgstpbutils-1.0.so \\
/fb/lib/libgstriff-1.0.so \\
/fb/lib/libgstrtp-1.0.so \\
/fb/lib/libgstrtsp-1.0.so \\
/fb/lib/libgstsdp-1.0.so \\
/fb/lib/libgsttag-1.0.so \\
/fb/lib/libgstvideo-1.0.so \\
/fb/lib/pkgconfig/gstreamer-allocators-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-app-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-audio-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-fft-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-pbutils-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-plugins-base-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-riff-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-rtp-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-rtsp-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-sdp-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-tag-1.0.pc \\
/fb/lib/pkgconfig/gstreamer-video-1.0.pc

exec=\\
/fb/bin/gst-device-monitor-1.0 \\
/fb/bin/gst-discoverer-1.0 \\
/fb/bin/gst-play-1.0
/fb/lib/libgstallocators-1.0.so.0 \\
/fb/lib/libgstapp-1.0.so.0 \\
/fb/lib/libgstaudio-1.0.so.0 \\
/fb/lib/libgstfft-1.0.so.0 \\
/fb/lib/libgstpbutils-1.0.so.0 \\
/fb/lib/libgstriff-1.0.so.0 \\
/fb/lib/libgstrtp-1.0.so.0 \\
/fb/lib/libgstrtsp-1.0.so.0 \\
/fb/lib/libgstsdp-1.0.so.0 \\
/fb/lib/libgsttag-1.0.so.0 \\
/fb/lib/libgstvideo-1.0.so.0 \\
/fb/lib/gstreamer-1.0/libgstadder.so \\
EOF
test $ALSA && echo /fb/lib/gstreamer-1.0/libgstalsa.so \\ >> $DESTDIR/fb/share/pkgconfig/gst-plugins-base-build.pc
cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-base-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstapp.so \\
/fb/lib/gstreamer-1.0/libgstaudioconvert.so \\
/fb/lib/gstreamer-1.0/libgstaudiorate.so \\
/fb/lib/gstreamer-1.0/libgstaudioresample.so \\
/fb/lib/gstreamer-1.0/libgstaudiotestsrc.so \\
/fb/lib/gstreamer-1.0/libgstencodebin.so \\
/fb/lib/gstreamer-1.0/libgstgio.so \\
EOF
test $TREMOR && echo /fb/lib/gstreamer-1.0/libgstivorbisdec.so \\ >> $DESTDIR/fb/share/pkgconfig/gst-plugins-base-build.pc
cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-base-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstogg.so \\
/fb/lib/gstreamer-1.0/libgstpango.so \\
/fb/lib/gstreamer-1.0/libgstplayback.so \\
/fb/lib/gstreamer-1.0/libgstsubparse.so \\
/fb/lib/gstreamer-1.0/libgsttcp.so \\
/fb/lib/gstreamer-1.0/libgsttypefindfunctions.so \\
/fb/lib/gstreamer-1.0/libgstvideoconvert.so \\
/fb/lib/gstreamer-1.0/libgstvideorate.so \\
/fb/lib/gstreamer-1.0/libgstvideoscale.so \\
/fb/lib/gstreamer-1.0/libgstvideotestsrc.so \\
/fb/lib/gstreamer-1.0/libgstvolume.so \\
/fb/lib/gstreamer-1.0/libgstvorbis.so
EOF

exit
--- gst-plugins-base-1.6.4.orig/configure.ac
+++ gst-plugins-base-1.6.4/configure.ac
@@ -688,7 +688,7 @@
 dnl *** pango ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_PANGO, true)
 AG_GST_CHECK_FEATURE(PANGO, [Pango font rendering], pango, [
-  AG_GST_PKG_CHECK_MODULES(PANGO, pango >= 1.22.0  pangocairo >= 1.22.0)
+  AG_GST_PKG_CHECK_MODULES(PANGO, pango >= 1.16.0  pangocairo >= 1.16.0)
 ])
 
 dnl *** theora ***
--- gst-plugins-base-1.6.4.orig/ext/pango/gstbasetextoverlay.c
+++ gst-plugins-base-1.6.4/ext/pango/gstbasetextoverlay.c
@@ -316,7 +316,11 @@
     g_mutex_lock (klass->pango_lock);
   fontmap = pango_cairo_font_map_get_default ();
   klass->pango_context =
+#if PANGO_VERSION_CHECK(1,22,0)
       pango_font_map_create_context (PANGO_FONT_MAP (fontmap));
+#else
+      pango_cairo_font_map_create_context (PANGO_CAIRO_FONT_MAP (fontmap));
+#endif
   pango_context_set_base_gravity (klass->pango_context, PANGO_GRAVITY_SOUTH);
   if (klass->pango_lock)
     g_mutex_unlock (klass->pango_lock);
--- gst-plugins-base-1.6.4.orig/ext/pango/gsttextrender.c
+++ gst-plugins-base-1.6.4/ext/pango/gsttextrender.c
@@ -201,7 +201,11 @@
 
   fontmap = pango_cairo_font_map_get_default ();
   klass->pango_context =
+#if PANGO_VERSION_CHECK(1,22,0)
       pango_font_map_create_context (PANGO_FONT_MAP (fontmap));
+#else
+      pango_cairo_font_map_create_context (PANGO_CAIRO_FONT_MAP (fontmap));
+#endif
   g_object_class_install_property (G_OBJECT_CLASS (klass), PROP_FONT_DESC,
       g_param_spec_string ("font-desc", "font description",
           "Pango font description of font "
--- gst-plugins-base-1.6.4.orig/gst-libs/gst/fft/fft.h
+++ gst-plugins-base-1.6.4/gst-libs/gst/fft/fft.h
@@ -22,7 +22,6 @@
 #ifndef __GST_FFT__H__
 #define __GST_FFT__H__
 
-#include <gst/fft/gstfft.h>
 #include <gst/fft/gstffts16.h>
 #include <gst/fft/gstffts32.h>
 #include <gst/fft/gstfftf32.h>
--- gst-plugins-base-1.6.4.orig/gst-libs/gst/riff/riff.h
+++ gst-plugins-base-1.6.4/gst-libs/gst/riff/riff.h
@@ -22,7 +22,6 @@
 #ifndef __GST_RIFF_H__
 #define __GST_RIFF_H__
 
-#include <gst/riff/riff-ids.h>
 #include <gst/riff/riff-media.h>
 #include <gst/riff/riff-read.h>
 
--- gst-plugins-base-1.6.4.orig/gst-libs/gst/rtsp/rtsp.h
+++ gst-plugins-base-1.6.4/gst-libs/gst/rtsp/rtsp.h
@@ -22,7 +22,6 @@
 #ifndef __GST_RTSP__H__
 #define __GST_RTSP__H__
 
-#include <gst/rtsp/gstrtsp.h>
 #include <gst/rtsp/gstrtspconnection.h>
 #include <gst/rtsp/gstrtspdefs.h>
 #include <gst/rtsp/gstrtsp-enumtypes.h>
--- gst-plugins-base-1.6.4.orig/gst-libs/gst/sdp/sdp.h
+++ gst-plugins-base-1.6.4/gst-libs/gst/sdp/sdp.h
@@ -22,7 +22,6 @@
 #ifndef __GST_SDP__H__
 #define __GST_SDP__H__
 
-#include <gst/sdp/gstsdp.h>
 #include <gst/sdp/gstsdpmessage.h>
 #include <gst/sdp/gstmikey.h>
 
--- gst-plugins-base-1.6.4.orig/tests/examples/fft/fftrange.c
+++ gst-plugins-base-1.6.4/tests/examples/fft/fftrange.c
@@ -18,10 +18,7 @@
  */
 
 #include <stdio.h>
-#include <gst/fft/gstffts16.h>
-#include <gst/fft/gstffts32.h>
-#include <gst/fft/gstfftf32.h>
-#include <gst/fft/gstfftf64.h>
+#include <gst/fft/fft.h>
 
 /* effectively max range seems to be 1/4 of what it should be */
 
